### PR TITLE
Adjust to the new key-value logging regime

### PR DESCRIPTION
**Pull request by bensigelman**
(As proposed in https://github.com/opentracing/opentracing-go/pull/108)

cc: @yurishkuro @tschottdorf

### CODE REVIEW

File: span.go [hunk 1]
```diff
@@ -88,6 +90,35 @@ func (s *spanImpl) SetTag(key string, value interface{}) opentracing.Span {
 	return s
 }
 
+func (s *spanImpl) LogKV(keyValues ...interface{}) {
+	if len(keyValues)%2 != 0 {
+		s.LogFields(log.Error(fmt.Errorf("Non-even keyValues len: %v", len(keyValues))))
+		return
+	}
+	fields, err := log.InterleavedKVToFields(keyValues...)
```
Comment (bensigelman) [inline span.go]: (take advantage of this new helper)

File: span.go [hunk 2]
```diff
@@ -88,6 +90,35 @@ func (s *spanImpl) SetTag(key string, value interface{}) opentracing.Span {
 	return s
 }
 
+func (s *spanImpl) LogKV(keyValues ...interface{}) {
+	if len(keyValues)%2 != 0 {
+		s.LogFields(log.Error(fmt.Errorf("Non-even keyValues len: %v", len(keyValues))))
```
Comment (yurishkuro) [inline span.go]: any reason not to have the helper method return this error?

Comment (bensigelman) [inline span.go]: it does -- forgot to remove here (done now)

**Comment by RaduBerinde** [comment]
We also need to adjust NetTraceIntegrator to handle `EventLogFields`, no?

**Comment by bensigelman** [comment]
@RaduBerinde that makes sense (kinda), though I have never used the NetTraceIntegrator myself... if I broke a build somewhere, let me know and I'll look into this and fix. Otherwise, I can either "put it on my list" or you can include the (presumably trivial) change in your outstanding PR as a one-off commit.

**Comment by RaduBerinde** [comment]
Yes, I can do it, it should be an easy change. I don't think any build is broken, just that messages that use `LogKV`/`LogFields` won't show up. The only thing I'm not sure is how to format a `LogRecord` into a single readable string.

**Pull request merged**